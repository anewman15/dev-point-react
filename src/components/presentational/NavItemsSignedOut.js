import { Link } from 'react-router-dom';

function NavItemsSignedOut() {
  return (
    <div>
      <Link to="/sign_up" className="button is-primary"><strong>Sign up</strong></Link>
      <Link to="/login" className="button is-light">Log in</Link>
    </div>
  );
}

export default NavItemsSignedOut;
