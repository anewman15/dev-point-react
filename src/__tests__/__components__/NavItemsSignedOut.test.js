import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavItemsSignedOut from '../../components/presentational/NavItemsSignedOut';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('NavItemsSignedOut', () => {
  it('renders the NavItemsSignedOut component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavItemsSignedOut />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
