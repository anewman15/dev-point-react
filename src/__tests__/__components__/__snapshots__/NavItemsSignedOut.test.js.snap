// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItemsSignedOut renders the NavItemsSignedOut component to the dom 1`] = `
Array [
  <div
    className="navbar-item"
  >
    <a
      className="btn btn-success"
      href="/login"
      onClick={[Function]}
    >
      Log in
    </a>
  </div>,
  <div
    className="navbar-item"
  >
    <a
      className="btn btn-secondary"
      href="/sign_up"
      onClick={[Function]}
    >
      <strong>
        Sign up
      </strong>
    </a>
  </div>,
]
`;
